code/datums/actions/item_action.dm

```diff
@@ -82,3 +82,14 @@
 			current_action_vis_obj = autoburst
 	button.vis_contents += current_action_vis_obj
 	action_firemode = holder_gun.gun_firemode
+
+/datum/action/item_action/aim_mode/update_button_icon()
+	button.overlays.Cut()
+	name = "Take Aim"
```

Comment:
no, put it on the datum

Reply:
I am not sure if it is ok, but I declared it in the datum using "var/aim_mode_name" rather than "name". Otherwise if it is using "name", this var gets assigned the name of the item in New() rather than the name set in the datum.